src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpAuthenticationManager.cs

```diff
@@ -11,8 +11,10 @@ internal static class SmtpAuthenticationManager
 
         static SmtpAuthenticationManager()
         {
+#if !NO_NTAUTHENTICATION
             Register(new SmtpNegotiateAuthenticationModule());
             Register(new SmtpNtlmAuthenticationModule());
+#endif
```

Comment:
Alternatively this could be split into `.NTAuth.cs` and `.NoNTAuth.cs` partial files. It felt like an overkill but I can do it if consistency is favored.